src/test/components/ProfileCallTreeView.test.js

```diff
@@ -488,3 +492,52 @@ describe('ProfileCallTreeView/end-to-end', () => {
     );
   });
 });
+
+describe('ProfileCallTreeView with JS Allocations', function() {
+  function setup() {
+    const { profile } = getProfileWithJsAllocations();
+    const store = storeWithProfile(profile);
+    const renderResult = render(
+      <Provider store={store}>
+        <ProfileCallTreeView hideThreadActivityGraph={true} />
+      </Provider>
+    );
+
+    return { profile, ...renderResult, ...store };
+  }
+
+  it('can switch to JS allocations and back to timing', function() {
+    const { getByText, getState } = setup();
+
+    // It starts out with timing.
+    expect(getCallTreeSummaryStrategy(getState())).toEqual('timing');
+
+    // It switches to JS allocations.
+    getByText('JavaScript Allocations').click();
```

Comment:
optional: I'd use `fireEvent.click(getByText('JavaScript Allocations'))` which sends a real `MouseEvent` instead of relying on the DOM click functions doing the right thing.

That said, MDN says:
```
When click() is used with supported elements (such as an <input>), it fires the element's click event. This event then bubbles up to elements higher in the document tree (or event chain) and fires their click events.
```
So I'd say it's fine... if JSdom implements this correctly, which it probably does, otherwise tests would be red :)

Reply:
I'd lean to keeping it the same for now.